### PR TITLE
bail out of main() if use_juju_for_storage=True and unsupported

### PR DESCRIPTION
**Pull request by chipaca**
fixes: #344

### CODE REVIEW

**Review by justinmclark — APPROVED** [review]
This looks good to me. For my own understanding, `ops.storage._JujuStorageBackend.is_available()` is equivalent to checking if `state-get` exists?

File: ops/main.py [hunk 1]
```diff
@@ -335,6 +335,11 @@ def main(charm_class: ops.charm.CharmBase, use_juju_for_storage: bool = None):
 
     charm_state_path = charm_dir / CHARM_STATE_FILE
 
+    if use_juju_for_storage and not ops.storage._JujuStorageBackend.is_available():
```
Comment (facundobatista) [inline ops/main.py:338]: We shouldn't use storage's private attributes from main.

**Review by facundobatista — CHANGES_REQUESTED** [review]

File: ops/main.py (hunk 1 above)
Comment (chipaca) [reply ops/main.py]: fixed

**Comment by chipaca** [comment]
> `ops.storage._JujuStorageBackend.is_available()` is equivalent to checking if `state-get` exists?

not sure *equivalent* is the right word :-) is_available is implemented by checking whether state-get exists (but it could also use `state-set` with next to no changes, or check the version if we were that confident).

**Review by facundobatista — APPROVED** [review]
Great, thanks!

**Pull request merged**